### PR TITLE
Bump `fundraiser`

### LINKED ISSUE #445: Fundraiser: harness.rs has outdated comments
### Motivation

All of the comments [here](https://github.com/FuelLabs/sway-applications/blob/0132a4a89bb0f9c1a5d016db609adead87c43626/fundraiser/project/fundraiser-contract/tests/harness.rs#L1-L11) are outdated / implemented. Remove them.

### LINKED ISSUE #656: Bump `fundraiser`
To:

- `forc 0.42.1`
- `fuel-core 0.18.3`
- `fuels-rs 0.42.0`

### PR DESCRIPTION
**Pull request by K1-R1**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)
- Documentation

## Changes

The following changes have been made:

- Bump application to `forc 0.42.1` and `fuel-core 0.18.3`
- Bump test harness to `fuels 0.42.0`

## Related Issues

Closes #656 
Closes #445

### CODE REVIEW

File: fundraiser/project/Forc.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [workspace]
 members = [
   "./contracts/fundraiser-contract",
-  "./contracts/fundraiser-contract/tests/artifacts/asset",
+  "./contracts/test-artifacts/asset",
```
Comment (Braqzen) [inline fundraiser/project/Forc.toml:4]: As mentioned in other PRs, this should either be put back or removed and replaced with functionality from the SDK

Comment (K1-R1) [reply fundraiser/project/Forc.toml:4]: This is required due to changes with nested projects and forc workspaces. An error is produced if the test artifacts are kept in their old places.

Comment (K1-R1) [reply fundraiser/project/Forc.toml:4]: I've re-written to use the SDK

**Review by Braqzen — COMMENTED** [review]
Same as the DAO, remove `rand` and use `AssetId::from()` to create a static value

**Review by Braqzen — APPROVED** [review]

**Pull request merged**